### ISSUE TITLE
Most frequent searches with zero results all include "visibility_search=1" or "include_in_menu=1" 

### ISSUE THREAD

**Issue by rossbearman** [algolia/algoliasearch-magento#1022]
Hi,

A client using the Magento integration is wondering why all of the frequent searches for categories or products that return zero results (as shown by the Algolia Usage report) have either "visibility_search=1" (products) and "include_in_menu=1" (categories) at the end of the query.

I assumed this was related to some sort of filtering and isn't actually part of the query entered or shown to the user, however the module was set to show all products visible in the category by default, not just in search, so the "visibility_search=1" doesn't make as much sense. Just wanted to confirm this and better understand the results they're seeing. Thanks.

Algolia Magento version: 1.13.0

**Comment by damcou** [comment]
Hi @rossbearman  !

In catalogsearch page, the attribute "visibility_search=1" is indeed included to make sure that only products configured in Magento as "visible in search" are returned. (visible in search (3) + both in catalog/ in search (4) )

In category pages, its the same behaviour but with "visibility_catalog" (visible in catalog (2) + both in catalog/ in search (4) )

During indexing, visibility_search and visibility_catalog are added to product indices for every records.

Examples : 
Product A (visibility "not visible (1)"  in Magento)  => not indexed
Product B (visibility "in catalog (2)" in Magento) => indexed with visibility_catalog = 1 and visibility_search = 0 
Product C(visibility "in search (3)" in Magento) => indexed with visibility_catalog = 0 and visibility_search = 1 
Product D(visibility "both (4)" in Magento) => indexed with visibility_catalog = 1 and visibility_search = 1 

Products B and D will appear in categories if Algolia is activated for those pages.
Products C and D will appear in catalogsearch page.

In conclusion, those 2 attributes are added to simulate the "visibility behaviour" of Magento. 

Hope this will help.

Best regards.

**Comment by rossbearman** [comment]
Thanks @damcou, that's what I assumed and explains it perfectly.
